tasks/stl/voroshilov_v_convex_hull_components/src/chc.cpp

```diff
@@ -0,0 +1,416 @@
+#include "../include/chc.hpp"
+
+#include <algorithm>
+#include <atomic>
+#include <cmath>
+#include <cstddef>
+#include <stack>
+#include <thread>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+using namespace voroshilov_v_convex_hull_components_stl;
+
+Pixel::Pixel(int y_param, int x_param) : y(y_param), x(x_param), value(0) {}
+Pixel::Pixel(int y_param, int x_param, int value_param) : y(y_param), x(x_param), value(value_param) {}
+
+bool Pixel::operator==(const int value_param) const { return value == value_param; }
+bool Pixel::operator==(const Pixel& other) const { return (y == other.y) && (x == other.x); }
+
+Image::Image(int hght, int wdth, std::vector<int> pxls) {
+  height = hght;
+  width = wdth;
+  pixels.resize(height * width);
+
+  int num_threads = static_cast<int>(std::thread::hardware_concurrency());
+  int chunk = (height + num_threads - 1) / num_threads;
```

Comment:
```suggestion

  int num_threads = static_cast<int>(ppc::util::GetPPCNumThreads());
  int chunk = (height + num_threads - 1) / num_threads;
```
cause in some tests it may work incorrect

Reply:
I agree. Fixed it